### PR TITLE
Adjust zbmath fetcher

### LINKED ISSUE #7202: Add support to add an entry from zbmath.org via its zbMATH id
JabRef supports adding entries via different kind of ids (https://docs.jabref.org/collect/add-entry-using-an-id). Right now, it is not possible to add an entry from [zbmath.org](https://zbmath.org) via its zbMATH id, however it is possible to add entries by searching its database (https://docs.jabref.org/collect/import-using-online-bibliographic-database). 

I would like to implement the possibility to add entries directly via their zbMATH id. I think this can be done quite easily by adjusting the zbMATH-Fetcher. I think this would be interesting as zbmath will become open access from January 2021 and thus no subscribtion for it is needed anymore.

### PR DESCRIPTION
**Pull request by ibe-314**
I implemented an IdBasedParserFetcher for the zbMATH fetcher [#7202]. 

fixes #7202 

zbMATHidfetcher

Running tests with gradle does not work locally on my computer, thus I create a draft pull request to run the tests. As I only changed the zbMATH fetcher, only zbMATHTest.java should be affected 

- [ x] Change in CHANGELOG.md described (if applicable)
- [ x] Tests created for changes (if applicable)
- [ x] Manually tested changed features in running JabRef (always required)
- [ x] Screenshots added in PR description (for UI changes)
- [ x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository. -> Will create a pull request to update documentation.

### CODE REVIEW

**Comment by ibe-314** [comment]
> 
> 
> Looks already good so far, I think the only difference is now that the search returns more than one entry:
> 
> org.jabref.logic.importer.fetcher.ZbMATHTest
> 
> Test searchByQueryFindsEntry() FAILED

I think this failure is not due to my changes. The searchByQueryFindsEntry() tests the zbmath SearchBasedParserFetcher implementation that I have not changed. This test was disabled on CIServer and I enabled it because no subscribtion to zbMATH is needed anymore. I think the FAILED test was just not visible before.

The problem here is that the search query `an:0507.57010` is transformed into ` "0507.57010"  ` before it is passed to the SearchBasedParserFetcher. 
I added a print statement in the function getURLforQuery at the function in the zbMATH Fetcher [https://github.com/JabRef/jabref/blob/master/src/main/java/org/jabref/logic/importer/fetcher/ZbMATH.java](https://github.com/JabRef/jabref/blob/master/src/main/java/org/jabref/logic/importer/fetcher/ZbMATH.java. Then I started jabref using gradle and tested the web search using the gui. My input is `an:0507.57010`. However, the function getURLforQuery does not recive `an:0507.57010` as its query parameter but ` "0507.57010"  `. So somewhere the input query is transformed. I also tested other structured queries for zbmath (e.g. `au:Berge ti:"Graph Theory"` is transformed to ` "Graph Theory" "Berge"  `) and it always transforms the query, which is not so good in this case as no structured query is possible. Is there a way to directly pass the user input string to the query parameter at getURLforQuery?

**Comment by tobiasdiez** [comment]
I think this problem comes from the recent improvements to support complex queries. The `an:` query part is not supported and thus removed. @DominikVoigt can probably say more about this.

The test should be changed either way to search by say using the title and author, to better reflect how a user would work with the fetcher.

**Comment by ibe-314** [comment]
Thank you for the information.
I did not know about the new search query. Is it described anywhere? I tried the query `author:"Claude Berge"` and it returned also documents that are not authored by Claude Berge. 

Back to this PR: I added a test called searchByIdFindsEntry() at org.jabref.logic.importer.fetcher.ZbMATHTest to test the implementation of the IdBasedParserFetcher. When I look at the logs of Fetcher Test, then this test does not appear, thus I guess that it was successful.

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 
 ### Added
 
+- We implemented an IdBasedParserFetcher for zbmath.
```
Comment (tobiasdiez) [inline CHANGELOG.md:14]: The changelog is mostly for our users. Could you thus provide a more "understandable" explanation. Thanks!

**Review by tobiasdiez — CHANGES_REQUESTED** [review]
I agree, this is not really related to this PR.

Codewise this looks good to me. Please improve the changelog entry a bit, then this is ready to go from my side. Thanks again!

File: CHANGELOG.md (hunk 1 above)
Comment (ibe-314) [reply CHANGELOG.md]: I changed the changelog.

**Review by Siedlerchr — APPROVED** [review]

**Comment by Siedlerchr** [comment]
Thanks a lot for your contribution! That will be very useful for users in that field of research.

**Pull request merged**